packages/diagrams/frontend/sirius-components-diagrams/src/sprotty/views/InsideLabelView.tsx

```diff
@@ -0,0 +1,107 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+/** @jsx svg */
+/** @jsxRuntime classic */
+import { setAttr, setNamespace, SLabelView, svg } from 'sprotty';
+import { getSubType } from 'sprotty-protocol';
+const preventRemovalOfUnusedImportByPrettier = svg !== null;
+
+/**
+ * The view used to display inside labels.
+ *
+ * @arichard
+ */
+export class InsideLabelView extends SLabelView {
+  // @ts-ignore
+  render(label) {
+    const { color, bold, underline, strikeThrough, italic, fontSize, iconURL, opacity } = label.style;
+
+    const width: number = label.parent.size.width - label.position.y * 2;
+    const height: number = label.parent.size.height - label.position.x * 2;
+
+    // The font-family is hardcoded to match with the backend compute bounds algo.
+    const textStyle = {
+      width: width + 'px',
+      height: height + 'px',
+      color: color,
+      opacity: opacity,
+      'font-size': fontSize + 'px',
+      'font-family': 'Arial, Helvetica, sans-serif',
+      'font-weight': 'normal',
+      'font-style': 'normal',
+      'overflow-wrap': 'anywhere',
+      'white-space': 'break-spaces',
+      display: 'flex',
+      'flex-wrap': 'no-wrap',
+    };
+    if (bold) {
+      textStyle['font-weight'] = 'bold';
+    }
+    if (italic) {
+      textStyle['font-style'] = 'italic';
+    }
+    if (underline) {
+      textStyle['text-decoration'] = 'underline';
+    }
+    if (strikeThrough) {
+      if (textStyle['text-decoration'] === 'none') {
+        textStyle['text-decoration'] = 'line-through';
+      } else {
+        textStyle['text-decoration'] += ' line-through';
+      }
+    }
+    if (label.type.endsWith('-h_left')) {
+      textStyle['text-align'] = 'left';
+      textStyle['justify-content'] = 'flex-start';
+    } else if (label.type.endsWith('-h_right')) {
+      textStyle['text-align'] = 'right';
+      textStyle['justify-content'] = 'flex-end';
+    } else {
+      textStyle['text-align'] = 'center';
+      textStyle['justify-content'] = 'center';
+    }
+    if (label.type.includes('-v_top')) {
+      textStyle['align-items'] = 'flex-start';
+    } else if (label.type.includes('-v_bottom')) {
+      textStyle['align-items'] = 'flex-end';
+    } else {
+      textStyle['align-items'] = 'center';
+    }
+
+    const foreignObjectContents: JSX.Element = (
```

Comment:
`JSX.Element`? I'm really not sure about that. Here we are manipulating Snabbdom, not React's virtual DOM. The signature of the superclass `SLabelView` is:

```
export class SLabelView extends ShapeView {
    render(label: Readonly<SLabel>, context: RenderingContext): VNode | undefined {
        ...
    }
}
```

If you want to type those local variables, those should be `VNode` and not `JSX.Element`

Reply:
The type should indeed be `VNode` but snabbdom is such a wonderful piece of technology to work with that you can't easily specify that right now. I'll suggest you just remove the invalid typing since it would just confuse others.